variable.c

```diff
@@ -1968,13 +1948,16 @@ rb_copy_generic_ivar(VALUE clone, VALUE obj)
 
     rb_check_frozen(clone);
 
-    if (!FL_TEST(obj, FL_EXIVAR)) {
-        goto clear;
+    if (BUILTIN_TYPE(obj) <= T_MODULE) {
```

Comment:
Is `BUILTIN_TYPE(obj) <= ` a common thing? I can see any other occurrence of it, and it seems a bit risky to couple that with the current values of the various T_* enums.